Bring Angular Devtools Into angular/angular

AleksanderBodurri opened the pull request:
This PR brings in the [Angular DevTools](https://github.com/rangle/angular-devtools) codebase into the Angular Repo. It preserves the commit history of the original repository with a process described [here](https://docs.google.com/document/d/1I4F8aKYe285prD3aXnu3yEoW-3FlG5DeuMFXp2fnLME).

The Angular DevTools codebase has been nested under a `devtools` directory. I will be working on integrating DevTools' Unit and E2E tests with the framework repository's CI.

Remaining tasks before merge
- [x] Verify that all authors have signed the CLA (Done)

Tasks After merge
- Reenable commit-message format validator in CI. Disabled because commits from the Angular DevTools repo did not follow the 20 character minimum limit for commits. Had to temporarily disable until these commits are merged.
- Reenable the branch rebasing workflow in CI. Had to temporarily disable because the commit graph from Angular DevTools  and the commit graph for angular/angular would be disjoint except for the unrelated-histories merge commit, meaning rebasing onto master would fail in CI.
- Build devtools with local angular packages

AleksanderBodurri committed:
fix(devtools): remove relative positioning from profiler content; makes split pane gutter full height for profile details

renovate-bot committed:
build(devtools): update angular-framework to 2b6651d

AleksanderBodurri committed:
refactor(devtools): move barchart component to live under barchart visualizer

twerske committed:
fix(devtools): contrast on labels and links (rangle/angular-devtools#708)

AleksanderBodurri committed:
fix(devtools): pass in root element ng context as an LView instead of as a context object

renovate-bot committed:
build(devtools): update angular-cli to 9adab5e

renovate-bot committed:
build(devtools): update angular-framework to 835342f

renovate-bot committed:
build(devtools): update dependency webpack to v5.35.1

AleksanderBodurri committed:
fix(devtools): prevent link in property view body from closing panel header when clicked (rangle/angular-devtools#714)

mgechev committed:
fix(devtools): increase the width of the filter

sumitarora committed:
docs(devtools): adding overview documentation

sumitarora committed:
docs(devtools): adding component inspector documentation

sumitarora committed:
docs(devtools): adding router tree documentation

sumitarora committed:
docs(devtools): adding images and gifs for documentation

AleksanderBodurri committed:
fix(devtools): disable animations when running e2e testing to fix flaky tests

renovate-bot committed:
build(devtools): update dependency webpack to v5.36.0

sumitarora committed:
docs(devtools): Update issue templates

sumitarora committed:
docs(devtools): updating documentation

renovate-bot committed:
build(devtools): update angular-cli to d926849

renovate-bot committed:
build(devtools): update angular-framework to 7b6e6e3

sumitarora committed:
docs(devtools): updating documentation

mgechev committed:
feat(devtools): implement negation in the filtering query language

mgechev committed:
feat(devtools): don't show comment nodes by default

mgechev committed:
refactor(devtools): move different profiler implementations to separate files

mgechev committed:
refactor(devtools): move the profiler subscription to the native profiler

mgechev committed:
feat(devtools): implement output profiling

mgechev committed:
feat(devtools): improve output formatting

renovate-bot committed:
build(devtools): update dependency webpack to v5.36.1

mgechev committed:
fix(devtools): proper dev mode detection when app compiled with closure

renovate-bot committed:
build(devtools): update angular-framework to 0badd71

sumitarora committed:
docs(devtools): updating readme (rangle/angular-devtools#724)

* docs: updating readme

* docs: updating readme

sumitarora committed:
docs(devtools): updating documentation

renovate[bot] committed:
build(devtools): update angular-cli to 25c716f (rangle/angular-devtools#725)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/chrome to ^0.0.136 (rangle/angular-devtools#731)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

AleksanderBodurri committed:
feat(devtools): implement drag to scroll functionality to improve frame navigation (rangle/angular-devtools#716)

mgechev committed:
build(devtools): update scripts and manifest

mgechev committed:
fix(devtools): set proper icon sizes

mgechev committed:
refactor(devtools): use material icons in popups

AleksanderBodurri committed:
fix(devtools): disable router tree tab

Previously the router parser uses private APIs to create the visualization for the router tree tab. Now the router parser always returns an empty router tree, effectively disabling the router tree tab until the necessary framework APIs are developed.

mgechev committed:
fix(devtools): get all the property names including getters

This PR intentionally ignores `Symbol`s because their serialization and deserialization will include extra complexity.

The changes also improve the typings in the continuation function that's passed during serialization.

sumitarora committed:
fix(devtools): renaming master to main

sumitarora committed:
feat(devtools): updating to karma-coverage

sumitarora committed:
feat(devtools): updating to karma-coverage and setting clear context to true

sumitarora committed:
fix(devtools): adding coverage reportor to gitignore

sumitarora committed:
feat(devtools): adding info popup (rangle/angular-devtools#739)

* feat: adding info popup

* Update projects/ng-devtools/src/lib/devtools-tabs/devtools-tabs.component.html

Co-authored-by: Minko Gechev <mgechev@gmail.com>

twerske committed:
fix(devtools): doc links icon (rangle/angular-devtools#742)

mgechev committed:
test(devtools): fix broken integration test

mgechev committed:
fix(devtools): improve the styles in the meta popup

sumitarora committed:
test(devtools): updating todos

sumitarora committed:
docs(devtools): updating images

renovate-bot committed:
build(devtools): update angular-framework to 194aa1e

renovate-bot committed:
build(devtools): update dependency @types/chrome to ^0.0.137

renovate-bot committed:
build(devtools): update angular-cli to ece77fb

renovate-bot committed:
build(devtools): update dependency @types/jasmine to ~3.7.0

renovate-bot committed:
build(devtools): update angular-framework to 5b26a8c

renovate-bot committed:
build(devtools): update dependency @types/chrome to ^0.0.138

sumitarora committed:
feat(devtools): adding tooltips for icons

mgechev committed:
refactor(devtools): better naming and removal of log statement (rangle/angular-devtools#753)

AleksanderBodurri committed:
feat(devtools): focus search input when icon is clicked

Previously, clicking the search icon on the component explorer did nothing. Now it focuses search input on click.

AleksanderBodurri committed:
feat(devtools): expand node ancestors on component inspection

Previously if a component was inspected with the component inspector and that component was collapsed in the component explorer, the component would be selected but would not be viewable until the explorer was manually expanded.
Now ancestors are expanded on component inspection.

mgechev committed:
fix(devtools): preserve proper nesting in flattened node lists

Rather than naively filter the flattened list, preserve the nesting by
filtering the original tree.

mgechev committed:
refactor(devtools): remove log statements and update styles when no ng app found

mgechev committed:
fix(devtools): recognize app after reopening devtools

Previously, Angular DevTools did not recognize the Angular application the second time the user reopens the extension tab. Webpack was preventing us to invoke twice a module with the same name. This PR changes the `content-script` to a library, allowing us to invoke it multiple times.

mgechev committed:
fix(devtools): breadcrumb algorithm when there is no parent available

renovate-bot committed:
build(devtools): update angular-framework to 963a2a6

renovate-bot committed:
build(devtools): update dependency webpack to v5.37.0

renovate-bot committed:
build(devtools): update angular-cli to 4510d13

renovate-bot committed:
build(devtools): update dependency @types/chrome to ^0.0.139

mgechev committed:
build(devtools): consistent branding

mgechev committed:
fix(devtools): use the aio docs link rather than github

renovate-bot committed:
build(devtools): update angular-components to v12

renovate-bot committed:
build(devtools): update dependency ng-packagr to v12

renovate-bot committed:
build(devtools): update dependency @types/chrome to ^0.0.140

renovate-bot committed:
build(devtools): update angular-framework

mgechev committed:
fix(devtools): wrong label in the profiler

sumitarora committed:
docs(devtools): updating documentation

sumitarora committed:
docs(devtools): adding link to chrome web store and using yarn

renovate-bot committed:
build(devtools): update dependency webpack to v5.37.1

sumitarora committed:
docs(devtools): updating contributing doc

renovate-bot committed:
build(devtools): update angular-framework to 5d43cbd

renovate-bot committed:
build(devtools): update dependency @types/chrome to ^0.0.141

mgechev committed:
docs(devtools): update issue templates

renovate-bot committed:
build(devtools): update angular-cli to 61d5a44

AleksanderBodurri committed:
fix(devtools): prevent directive forest hooks from being initiated in production applications

markwhitfeld committed:
fix(devtools): don't access service prop unnecessarily

The code was requesting index 0 of each item in the LView before checking
if it is an array. This causes issues with services that only allow calls to
properties that are defined on the service (backed by a strict ES6 proxy).
A check if the type is Array first corrects this issue.

fixes issue rangle/angular-devtools#802

mgechev committed:
fix(devtools): update the prod mode detection to also consider VE apps

Fix rangle/angular-devtools#798

mgechev committed:
fix(devtools): improve dark mode styles when no ivy detected

amirrustam committed:
ci(devtools): fix flaky tests using cypress orb (rangle/angular-devtools#733)

* ci: refactor circle config to use cypress orb

* ci: skipping unit tests while validating e2e runs

* ci: fix karma chrome launches

* ci: investigating karma headless failure

* ci: iterating on failure

* ci: move environment to executor

* build: revert karma config changes

* ci: update karma config

Co-authored-by: mgechev <mgechev@gmail.com>

mgechev committed:
style(devtools): add call signatures

mgechev committed:
release(devtools): v1.0.1

renovate[bot] committed:
build(devtools): update dependency ts-node to v10 (rangle/angular-devtools#812)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to c54972c (rangle/angular-devtools#808)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/chrome to ^0.0.142 (rangle/angular-devtools#819)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to 729243c (rangle/angular-devtools#810)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

markwhitfeld committed:
fix(devtools): skip node if there is no associated TViewData (rangle/angular-devtools#804)

* fix(backend): skip node if there is no associated TViewData

The code was assuming that the TViewData item at the same index as the node
in LView would contain a data entry. This value can be `null` (which is the
case in our app), which causes a catastrophic failure in the dev tool. By
checking for null, the node is skipped because it cannot determine the
associated component or directive and the dev tool does not die.

fixes issue rangle/angular-devtools#801

* chore: style tweak - add early return to avoid nesting

renovate[bot] committed:
build(devtools): update angular-cli to 151d244 (rangle/angular-devtools#822)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 78ceca3 (rangle/angular-devtools#823)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

mgechev committed:
fix(devtools): make sure we update the property explorer on state change (rangle/angular-devtools#821)

Fix rangle/angular-devtools#786

With this PR now we consider not only the property name but also the
value of properties in the differ comparison.

The change also contains e2e and unit tests.

mgechev committed:
fix(devtools): delayed angular detection with app_initializer (rangle/angular-devtools#827)

This PR sets a timeout for repetitive check if there's an Angular app
on the page. Fix rangle/angular-devtools#826.

mgechev committed:
fix(devtools): show properly dynamically inserted content with DOM APIs (rangle/angular-devtools#828)

Fix rangle/angular-devtools#791

Build the render tree starting from the root node of the application.
This fix is applicable only for v12+ apps that are using the latest
debugging APIs.

mgechev committed:
release(devtools): v1.0.2

renovate[bot] committed:
build(devtools): update angular-framework to ab9c21b (rangle/angular-devtools#829)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/chrome to ^0.0.143 (rangle/angular-devtools#830)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency tsickle to ^0.40.0 (rangle/angular-devtools#831)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency webpack to v5.38.1 (rangle/angular-devtools#832)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

DaanDeSmedt committed:
docs(devtools): small type on README.md (rangle/angular-devtools#836)

A small type fixed on the README.md file

renovate[bot] committed:
build(devtools): update angular-cli to 4acdfec (rangle/angular-devtools#835)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

sumitarora committed:
fix(devtools): incorrect input names

sumitarora committed:
refactor(devtools): removing input & output rename lint rules, adding examples for the same

sumitarora committed:
test(devtools): updating tests for input & output property names

sumitarora committed:
test(devtools): adding e2e tests for inputs, outputs & properties

renovate[bot] committed:
build(devtools): update angular-cli to 26eb107 (rangle/angular-devtools#842)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/chrome to ^0.0.144 (rangle/angular-devtools#841)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to b5ab092 (rangle/angular-devtools#840)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 801e38a (rangle/angular-devtools#844)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to bdc3a6b (rangle/angular-devtools#843)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to 372f83f (rangle/angular-devtools#846)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

AleksanderBodurri committed:
fix(devtools): prevent mat button css from leaking out of property view header (rangle/angular-devtools#856)

Previously, the property view header component had mat button styling nested in an ng-deep selector. This broke some styling in the info popup whenever an item was selected in the component explorer (because the existence of a selected component causes the property view header tab to render and thus brings in the leaky component css).

Now the leaky styling has been removed, and the component css has been cleaned up.

AleksanderBodurri committed:
refactor(devtools): switch to sass use and new angular material theming API (rangle/angular-devtools#853)

Previously we were using sass import rules to bring in material and other styling.

Now we are using the use rule and have switched over to the new angular material theming API

renovate[bot] committed:
build(devtools): update dependency webpack to v5.39.1 (rangle/angular-devtools#859)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

AleksanderBodurri committed:
fix(devtools): use visible gutter size in calculation for split area flex basis' (rangle/angular-devtools#857)

After PR rangle/angular-devtools#618, an invisible gutter was introduced to increase the draggable area of the split pane gutter. The gutter size being used in the calculation for the split area flex-basis still referenced the size of the now invisible gutter, causing split areas to be smaller than they should be. This is most noticable in the the input/output/state expandable panels in the property explorer tab.

Now the visible gutter size is used in the flex-basis calculation instead of the invisible one.

AleksanderBodurri committed:
fix(devtools): update typing of differ objects to fit the new contract of TrackByFunction (rangle/angular-devtools#862)

Previously `DefaultIterableDiffer` was able to infer the type of it's generic by reading the type contract defined by the user defined trackBy function. This was because `TrackByFunction` can be specified with a generic that is used to type its second argument. If this generic is not provided, like in this case, typescript infers this type from the function contract directly. In this case this inferred type was `FlatNode`. This inference is then propagated to `DefaultIterableDiffer`, which is why we were not seeing this error previously.

A change in the framework made it so the second argument of `TrackByFunction` is typed as a generic that extends the generic passed into `TrackByFunction`. Since we had not previously passed in this generic to `TrackByFunction`, this type was inferred as an extension of `unknown`, causing a type error when passed as an argument to the `DefaultIterableDiffer` constructor.

Now the functions being used are typed directly as `TrackByFunction<FlatNode>`

For clarity, the `DefaultIterableDiffer` generic is now also typed as `FlatNode`.

See https://github.com/angular/angular/pull/41995 for the framework change.

renovate[bot] committed:
build(devtools): update dependency tsickle to ^0.43.0 (rangle/angular-devtools#858)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to f480d2c (rangle/angular-devtools#854)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to cf52ccc (rangle/angular-devtools#847)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 8dd79b2 (rangle/angular-devtools#869)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-components to ~12.1.0 (rangle/angular-devtools#867)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/chrome to ^0.0.145 (rangle/angular-devtools#851)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency webpack to v5.40.0 (rangle/angular-devtools#863)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency wait-on to v6 (rangle/angular-devtools#870)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 8dbf421 (rangle/angular-devtools#871)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency webpack to v5.41.1 (rangle/angular-devtools#872)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 5116711 (rangle/angular-devtools#873)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 3634d08 (rangle/angular-devtools#875)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to d409506 (rangle/angular-devtools#865)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to f40f67d (rangle/angular-devtools#883)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency ts-node to ~10.1.0 (rangle/angular-devtools#884)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to ce30802 (rangle/angular-devtools#885)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to 8da208e (rangle/angular-devtools#886)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to 17947b8 (rangle/angular-devtools#887)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency cypress to v8 (rangle/angular-devtools#888)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to a2e6cd9 (rangle/angular-devtools#881)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to 58452bb (rangle/angular-devtools#890)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to d0d9b6a (rangle/angular-devtools#891)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/chrome to ^0.0.148 (rangle/angular-devtools#879)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency webpack to v5.46.0 (rangle/angular-devtools#877)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to a3e7465 (rangle/angular-devtools#894)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/chrome to ^0.0.149 (rangle/angular-devtools#898)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-cli to 3e34f45 (rangle/angular-devtools#893)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to e9d142c (rangle/angular-devtools#896)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 0cf5f52 (rangle/angular-devtools#905)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 7cc7528 (rangle/angular-devtools#906)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to e357b38 (rangle/angular-devtools#907)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 26f9407 (rangle/angular-devtools#908)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 1985e48 (rangle/angular-devtools#909)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to b372e0c (rangle/angular-devtools#911)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency ts-node to ~10.2.0 (rangle/angular-devtools#903)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-components to ~12.2.0 (rangle/angular-devtools#902)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency jasmine-core to ~3.9.0 (rangle/angular-devtools#876)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 854ecbc (rangle/angular-devtools#912)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/chrome to ^0.0.154 (rangle/angular-devtools#900)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update dependency @types/jasmine to ~3.8.0 (rangle/angular-devtools#882)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 3849283 (rangle/angular-devtools#913)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to 52c8b39 (rangle/angular-devtools#914)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update cypress orb to v1.29.0 (rangle/angular-devtools#901)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
build(devtools): update angular-framework to c6329d9 (rangle/angular-devtools#918)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

AleksanderBodurri committed:
build(devtools): set up bazel and bazel tooling

Marks the beginning of a migration over to building Angular devtools with Bazel. This commit brings in tooling from the canonical angular bazel example repository and includes the necessary dependencies in the package.json

AleksanderBodurri committed:
build(devtools): create bazel build rules for shared-utils and the protocol library

Allows shared-utils and protocol to be built with `bazel build projects/shared-utils` and `bazel build projects/protocol`. This lays down the necessary ground work to include these libraries as dependencies in other bazel build rules.

AleksanderBodurri committed:
build(devtools): migrate demo-no-zone to be built with bazel and allow it to be served with concatjs_devserver

Allows the demo-no-zone application to be built with bazel and a devserver run with `bazel run projects/demo-no-zone/src:devserver`

AleksanderBodurri committed:
build(devtools): migrate ng-devtools-backend to bazel

Allows the ng-devtools-backend to be built with bazel. Allows ng-devtools-backend to be used as a dependency in other bazel build rules.

AleksanderBodurri committed:
build(devtools): migrate ng-devtools library to bazel

Allows the ng-devtools library to be built with bazel. ng-devtools can now be used as a dependency in other bazel build rules.

AleksanderBodurri committed:
build(devtools): migrate demo app to bazel

Allows the demo app to be builtmwith bazel and a dev server started with `yarn start`.

AleksanderBodurri committed:
build(devtools): migrate shell-chrome to bazel

Allows shell-chrome to be built with bazel. Currently this is built in production mode with terser minified code and the appropriate manifest and content script files to make the chrome extension work properly.

AleksanderBodurri committed:
fix(devtools): add d3 and semver dsl umd modules to karma tests for ng-devtools and ng-devtools-backend

These modules are necessary because concatjs_devserver needs named amd/umd modules (https://bazelbuild.github.io/rules_nodejs/Concatjs.htmlrangle/angular-devtools#third-party-code)

renovate[bot] committed:
build(devtools): update dependency webpack to v5.52.0 (rangle/angular-devtools#897)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

mgechev committed:
build(devtools): invoke tslint directly via binary

mgechev committed:
style(devtools): fix linting errors

renovate-bot committed:
build(devtools): update angular-cli to 5666e34

renovate-bot committed:
build(devtools): update dependency @types/chrome to ^0.0.157

renovate-bot committed:
build(devtools): update dependency @types/jasmine to ~3.9.0

renovate-bot committed:
build(devtools): update angular-framework to 89ed9e0

renovate-bot committed:
build(devtools): update angular-cli to ef3abe2

renovate-bot committed:
build(devtools): update angular-framework to c6f5fe2

AleksanderBodurri committed:
build(devtools): create custom no sandbox chrome browser rule

Previously, running tests in CI with the chromium-local browser would cause the following error "Running as root without --no-sandbox is not supported".

Now the karma test suites run in a custom no sandbox browser.

mgechev commented:
@googlebot I consent

sumitarora commented:
@googlebot I consent

markwhitfeld commented:
@googlebot I consent.

AndrewKushnir commented:
@AleksanderBodurri thanks for working on this!

We've just discussed this PR during the framework team sync and we are trying to assess whether this code should live in the `angular/angular` repo or it should be located in it's own repo in Angular organization (for ex. `angular/devtools`). We'll have more discussion early next week and let you know. I just wanted to share it sooner in case there is some major work that you plan within the next couple days to integrate CI, etc.

renovate-bot committed:
build(devtools): update angular-framework to cf69793

renovate-bot committed:
build(devtools): update dependency webpack to v5.53.0

renovate-bot committed:
build(devtools): update angular-framework to a2c96f5

renovate-bot committed:
build(devtools): update angular-cli to bcf52c0

renovate-bot committed:
build(devtools): update angular-framework to ea0af7a

renovate-bot committed:
build(devtools): update angular-cli to b2b499f

renovate-bot committed:
build(devtools): update angular-cli to 86bed50

renovate-bot committed:
build(devtools): update angular-cli to 8932309

renovate-bot committed:
build(devtools): update angular-framework to 74ea521

renovate-bot committed:
build(devtools): update angular-cli to 9086b82

renovate-bot committed:
build(devtools): update angular-framework to c8eb298

renovate-bot committed:
build(devtools): update angular-framework to 71f8be3

renovate-bot committed:
build(devtools): update angular-framework to bd4d3e2

renovate-bot committed:
build(devtools): update angular-framework to 851fb04

renovate-bot committed:
build(devtools): update angular-framework to b8940cb

renovate-bot committed:
build(devtools): update dependency webpack to v5.58.0

renovate-bot committed:
build(devtools): update dependency webpack to v5.58.1

renovate-bot committed:
build(devtools): update dependency @types/chrome to ^0.0.159

mgechev committed:
feat(devtools): support trusted types for script injection

mgechev committed:
refactor(devtools): use safe operation to update the overlay

mgechev committed:
refactor(devtools): use local declaration of the change detection strategy enum

renovate-bot committed:
build(devtools): update dependency webpack to v5.58.2

renovate-bot committed:
build(devtools): update dependency ts-node to ~10.3.0

renovate-bot committed:
build(devtools): update dependency jasmine-core to ~3.10.0

AleksanderBodurri committed:
build(devtools): bump typescript version to keep angular devtools compatible with the angular compiler

Also installs the latest `@types/filesystem` because the transitive dependency from `@types/chrome` is not compatible with typescript 4.4 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54046

kamikopi committed:
fix(devtools): Support Electron (rangle/angular-devtools#849)

Following suggestion from the issue thread, created an empty polyfills
to the chrome.browserAction calls, which are not available in the electron.

renovate-bot committed:
build(devtools): update dependency webpack to v5.59.1

renovate-bot committed:
build(devtools): update angular-framework

AleksanderBodurri committed:
build(devtools): port over bazel tooling from angular/components

Previously the ts_library bazel rule was used to build angular devtools. This commit migrates to ts project and brings in some bazel tooling from angular/components to run bundle spec files for karma tests.

renovate-bot committed:
build(devtools): update angular-framework to e814a61

mgechev committed:
fix(devtools): handle ngFor and ngIf contexts in the profiler

Currently, we skip the time spent in evaluation of embedded view
templates because the passed context is either an NgIf or an NgForOf.

This change generalizes the implementation to work with views embedded
into a component.

mgechev committed:
release(devtools): v1.0.3

renovate-bot committed:
build(devtools): update angular-cli to 7c4c781

mgechev committed:
build(devtools): add .angular to gitignore

mgechev committed:
release(devtools): v1.0.4

AleksanderBodurri committed:
refactor(devtools): run buildifier on bazel files

AleksanderBodurri committed:
refactor(devtools): migrate cypress tests to use bazel/cypress

Previously we ran cypress directly. Now we use the `cypress_web_test` rule from rules_nodejs to run cypress through bazel.

//src:devapp is added to the `data` arg of `cypress_web_test`. This invalidates the bazel cache so that cypress tests rerun anytime our demo app (the app being e2e tested) changes.

AleksanderBodurri committed:
refactor(devtools): prepare codebase for migration to angular/angular repo

AleksanderBodurri committed:
refactor(devtools): merge rangle/angular-devtools into angular/angular

This is a merge commit that acts at the only common child of commits from the rangle/angular-devtools codebase and the angular/angular codebase.

This commit was created as the result of running `git merge` with the `--allow-unrelated-histories` flag.

AleksanderBodurri committed:
build(devtools): disable the "Rebase PR on target branch" circleci job so that the PR that merges the unrelated history of rangle/angular-devtools can be successfully merged into angular/angular

Usually this job would ensure that all PRs in angular/angular are running against the latest changes to angular/angular. In order to merge the unrelated history of the commits in the rangle/angular-devtools repo into angular/angular, we must temporarily disable this job.

AleksanderBodurri committed:
refactor(devtools): run ng-dev format on bazel files in devtools directory

AleksanderBodurri committed:
ci: add angular devtools to pullapprove config

Creates new section in pullapprove for angular devtools

AleksanderBodurri committed:
ci: disable commit message validation because commits from angular devtools don't have commit bodies

Commit messages from angular devtools did not follow the 20 char minimum limit for commit bodies rule. This commit disables the CI check for this so that the rest of CI can be validated. Will reenable once that is done.

AleksanderBodurri committed:
build: configure circleci for angular devtools ci

Sets up commands in circleci config to test angular devtools.

AleksanderBodurri committed:
ci: remove bazel cypress tests

Previously we were running Cypress with bazel in an effort to save on CI time when devtools dependencies did not change. This commit reverts to running Cypress manually, until we can revisit building devtools with local Angular packages to save CI time.

kamikopi commented:
@googlebot i consent

AleksanderBodurri committed:
refactor(devtools): dedupe some bazel tooling brought in from angular/components

Some tooling was brought in from angular/components to help with bundling spec files for Karma. This tooling has since been ported over to the dev-infra repo. This commit dedupes the tooling and instead depends on the angular/dev-infra-private repo.

AleksanderBodurri committed:
ci: add labels to angular devtools ci tasks

This commit also cleans up the .bazelrc file in the devtools directory

AleksanderBodurri committed:
build(devtools): use ng-dev stamp tooling to stamp the latest git SHA in devtools

Previously, we were using ngx-build-plus with a custom webpack config to determine the git SHA at runtime.

Now, after migrating to bazel and transfering to angular/angular, this commit uses ng-dev to stamp the latest git SHA directly in angular devtools' application environment.

This commit also removes the old webpack configs.

AleksanderBodurri committed:
ci: add angular devtools to the renovate configuration

Previously angular devtools was using its own renovate configuration to keep up to date. After transfering to angular/angular, we can use the renovate configuration in angular/angular to update our dependencies. In an end state we would build devtools with local Angular packages.

AleksanderBodurri committed:
test(devtools): fix some flakey cypress tests

Uses `cy.contains` as one command to optimize cypress' retry-ability mechanism

Affected tests are in:

`property-update.e2e.js` and `view-component-metadata.e2e.js`

josephperrott commented on .ng-dev/format.ts:
```diff
@@ -12,6 +12,7 @@ export const format: FormatConfig = {
   'clang-format': {
     'matchers': [
       '**/*.{js,ts}',
+      '!devtools/**',
```
We can do add it to the list of tasks after we merged if you prefer, but we should just run the formatter over all of the files in the devtools area before any work begins on the code once its in angular/angular as now would beb the least disruptive time for us to do so.

josephperrott commented on .pullapprove.yml:
```diff
@@ -1171,6 +1171,25 @@ groups:
+        - mgechev
+        - twerske
+        - josephperrott
+        - devversion
```
@mgechev Can you confirm the list of reviewers you were thinking for ownership here?

mgechev replied:
I'd suggest also adding @AleksanderBodurri

josephperrott commented on .circleci/config.yml:
```diff
@@ -727,6 +733,30 @@ jobs:
           no_output_timeout: 15m
+  test_angular_devtools:
+    executor:
+      name: default-executor
```
Rather than using the default executor and then manually installing the karma and cypress deps.  We should just create another executor definition here that uses the `cypress/browsers` image (ideally on v16 of node)

josephperrott reviewed (commented):
Overall I think we are in good shape, I left a few comments about a couple things.  But we should be good to land this in the next few days I think.  As you noted in the description there is a list of cleanup items to do after it lands, but I think we are in good shape for landing it.

josephperrott reviewed (changes requested):
One change we will need before we can land it though is that all of the the typescript files will need to contain the correct license header in them.

josephperrott commented:
To ensure the license header values are present, should add devtools to the tslint scope.

You can add it as included in `/tsconfig-tslint.json`

ladraum commented:
@googlebot i consent

devversion commented on devtools/.circleci/config.yml:
```diff
@@ -0,0 +1,91 @@
+version: 2.1
```
Is there an item in your cleanup list to remove this file?

devversion replied:
I see a lot of such files that should be removed, like the issue template etc. I assume though this is planned as a follow-up?

devversion commented on devtools/WORKSPACE:
```diff
@@ -0,0 +1,110 @@
+    patch_args = ["-p1"],
+    patches = [
+        # Updates @bazel/work dep to 4.0.0 inside rules_sass so it is compatible
+        "//:io_bazel_rules_sass.patch",
```
This workaround is no longer needed and can be removed. Angular components no longer does this.

devversion commented on devtools/packages.bzl:
```diff
@@ -0,0 +1,58 @@
+# List of MDC packages.
+MDC_PACKAGES = [
```
These constants can be removed from this sub-workspace. They were specific to Angular Material

devversion commented on devtools/packages.bzl:
```diff
@@ -0,0 +1,58 @@
+    ("@angular/platform-browser-dynamic", struct(entry_points = ["testing"])),
+    ("@angular/router", struct(entry_points = [])),
+    ("@angular/localize", struct(entry_points = ["init"])),
+]
```
Also note that this list should be the minimum for the devtools workspace. The more modules & entry-points listed here, the more builds are needed for the linked ESM bundles.

devversion commented on devtools/projects/demo-no-zone/src/app/BUILD.bazel:
```diff
@@ -0,0 +1,18 @@
+
+package(default_visibility = ["//visibility:public"])
+
+ng_ts_project(
```
Is there any reason `ts_project` is chosen over `ts_library`. `ts_library` would have various benefits:

* Consistency with the whole Angular organization
* Strict dependency checking
* Better incrementality as `ts_project` IIRC rebuilds dependant targets even if the types have not changed.

I think this can be a follow-up, but I think it's definitely something worth discussing, or did you already with @josephperrott?

devversion commented on devtools/projects/ng-devtools-backend/src/lib/directive-forest/BUILD.bazel:
```diff
@@ -0,0 +1,40 @@
+)
+
+ts_test_project(
+    name = "test-lib",
```
Just a heads up nit: Usually the convention is to use underscore as delimiter within targets. The only exception is the one matching the current package name (so that e.g. `bazel test devtools/a/b/c/test` would work).

devversion commented on devtools/tools/defaults.bzl:
```diff
@@ -0,0 +1,86 @@
+    web_test_args = {}
+    test_deps = kwargs.get("deps", [])
+
+    kwargs["tags"] = ["partial-compilation-integration"] + kwargs.get("tags", [])
```
This can be removed as you do not have any partial compilation integration tests set up. And I don't think that makes sense to do this in this workspace.

devversion commented on devtools/tools/defaults.bzl:
```diff
@@ -0,0 +1,86 @@
+    if not hasattr(kwargs, "browsers"):
+        kwargs["tags"] = ["native"] + kwargs.get("tags", [])
+        kwargs["browsers"] = [
+            # Note: when changing the browser names here, also update the "yarn test"
```
Nit: this comment only applied in the components repository and could be removed here.

devversion commented on devtools/tools/angular/index.bzl:
```diff
@@ -0,0 +1,78 @@
+        srcs = [":%s_linked_bundle" % target_name_base],
+        package = "@npm//@angular/%s" % pkg.name,
+        module_name = module_name,
+        subpath = target_name_base,
```
We should probably move this to the shared dev-infra code in a follow-up. I'll make a note.

The more I think about all of this though, I realize that technically all of this would not be needed if we'd consume the targets from the `angular` workspace directly. These are built in full compilation mode by default, so the linker would not be needed (just as a quick note)

devversion commented on devtools/src/environments/BUILD.bazel:
```diff
@@ -0,0 +1,14 @@
+    srcs = [
+        "environment.ts",
+    ],
+    tsconfig = "//:tsconfig.json",
```
For a follow-up: Isn't the `tsconfig` attribute here and everywhere else unnecessary? you specified a default in the Starlark macro.

AleksanderBodurri committed:
ci: add aleksanderbodurri to angular devtools pullapprove reviewers

AleksanderBodurri committed:
refactor(devtools): run ng-dev format on angular devtools files

Formats the entire devtools directory with the ng-dev formatting tool. Previously we relied on prettier, so this commit also remove prettier from devtools' dependencies.

AleksanderBodurri committed:
refactor(devtools): remove redundant project files

Previously angular devtools was in its own repository and so it had some project files related to github workflows, vscode settings, prettier configurations, etc.

Now, after devtools has been transfered to angular/angular, these files are redundant, and so this commit removes them.

AleksanderBodurri committed:
refactor(devtools): remove unneeded tsconfig keyword args for ts_project

These keyword args are not needed because `ts_project` and `ng_ts_project` already default to the tsconfig at the root of the devtools directory.

This commit removes these key word args from places where tsconfig was used as a keyword arg with the root level tsconfig as the target.

AleksanderBodurri committed:
refactor(devtools): clean up some of tooling brought in from angular/components

Removes MDC constants and unneeded angular package configs from packages.bzl.

Removes partial-compilation-integration from the karma web test tags because we don't perform any of partial compilation integration tests for devtools.

AleksanderBodurri committed:
refactor(devtools): remove inaccurate comment in bazelrc and defaults.bzl

Both of these comments were included accidentally after porting over bazel code from angular/angular and angular/components to devtools.

They are not relevant to devtools so this commit removes them.

AleksanderBodurri committed:
refactor(devtools): make bazel labels follow snake case convention

Previously, camel case was used for bazel labels in the devtools directory. This commit changes these labels to snake case except in the case where the label is identical to the directory name that the BUILD file is in.

AleksanderBodurri committed:
ci: change the devtools test executor to test-browser-executor and remove tasks in the test_angular_devtools job that installed test dependencies manually

This commit also disables cypress' video recording feature to save time on running Cypress tasks in CI.

AleksanderBodurri committed:
refactor(devtools): run tslint --fix on devtools codebase

This commit runs tslint --fix with the angular/angular tslint configuration on the files inside the devtools codebase.

Notably, the file-header rule in `tslint.json` was missing a default attribute. This commit adds that default attribute and sets it to the
license header that is present in all files in this repo. After running tslint --fix with this default added, this commit added the license header to all files in the devtools directory. Note for the reviewer: the automatically added license headers were added as comments with the "/*!" prefix. Since we want these comments removed in builds, and the rest of the codebase uses "/**", a simple find and replace was performed on the devtools directory to change these prefixes to "/**".

AleksanderBodurri committed:
refactor(devtools): remove rules_sass patch

Previously a workaround patch was needed because rules_sass didn't support v4 of rules_nodejs. This commit updates rules_sass to a version that does and removes the patch.

DaanDeSmedt commented:
@googlebot i consent

AleksanderBodurri committed:
refactor(devtools): add override to classes that extend abstract classes in the devtools directory

Previously the typescript override keyword was not on these methods. Now that devtools uses the tslint config in angular/angular (because of the no-implicit-override-abstract rule) we need to add these key words explicitly to method overrides. That is what this commit does.

JimMorrison723 commented on devtools/CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,12 @@
+
+As contributors and maintainers of the Angular DevTools project, we pledge to respect everyone who contributes by posting issues, updating documentation, submitting pull requests, providing feedback in comments, and any other activities.
+
+Communication through any of Angular's channels (GitHub, Gitter, IRC, mailing lists, Google+, Twitter, etc.) must be constructive and never resort to personal attacks, trolling, public or private harassment, insults, or other unprofessional conduct.
```
i am pretty sure it's okay to remove Google+ from the list...^^

mgechev replied:
We can remove this file completely.

AleksanderBodurri committed:
refactor(devtools): bring the angular devtools directory into the root bazel workspace

Previously devtools used a nested workspace for its bazel configurations. This meant framework dependencies were consumed via npm.

Now devtools is part of the root bazel directory that all other files in this codebase fall under. This allows us to build devtools using local angular packages, removing the need to consume these dependencies with npn. This is useful because we no longer have to update these dependencies with an automated tool like renovate, and our CI tests will always run against the most up to date framework packages.

devversion commented on .circleci/config.yml:
```diff
@@ -275,6 +277,7 @@ jobs:
           paths:
             - ./ng
             - ./bazel_repository_cache
+            - .cache/Cypress
```
I know this is deeply integrated already so far, but couple of important notes here:

1. This will increase the cache even more, with binaries. CircleCI will throttle the download if it exceeds 500MB IIRC and we were close to exceeding that before/might already do?

2. I think we need to make a discussion whether we'd want to use Cypress in this repo / have all team members install it, especially since it's just rarely used for the majority of people in the repo, right?

3. Cypress does not work with M1 natively and requires Rosseta 2 (which is something I'd like to avoid; since we want to support m1 natively in all repos): https://github.com/cypress-io/cypress/issues/4478.

I think ideally we'd run the Cypress tests within Bazel, and only download Cypress when actually needed, or we use tooling that is used elsewhere in the repo, if that is possible? (not sure if cypress is needed for the devtools)

cc. @josephperrott @mgechev

mgechev replied:
>I think we need to make a discussion whether we'd want to use Cypress in this repo / have all team members install it, especially since it's just rarely used for the majority of people in the repo, right?

I think we shouldn't have a requirement for all team members to install Cypress unless that's our e2e testing framework by choice. As part of the roadmap, we have migration from Protractor to a different e2e solution. I'd suggest:
1. Not run Cypress in CI for DevTools
2. Merge the PR
3. Migrate to the default e2e testing solution for the repo when we agree on it
4. Enable e2e tests for DevTools in CI

devversion replied:
@mgechev that sounds perfect to me 👍

josephperrott replied:
I think that makes sense to me as well.  As Paul mentioned, we definitely don't want to pay the cache cost if we don't have to, and I don't want to make every dev install on yarn install it when most wont use it.

devversion commented on devtools-bazel.rc:
```diff
@@ -0,0 +1,2 @@
+build --workspace_status_command="yarn -s ng-dev release build-env-stamp --mode=snapshot"
+build --stamp
```
Is this used somewhere?

devversion commented on devtools/.bazelignore:
```diff
@@ -0,0 +1,16 @@
+# NB: sematics here are not the same as .gitignore
+# see https://github.com/bazelbuild/bazel/issues/8106
+.git
+bazel-out
```
IIRC we can remove this file since this is no longer a nested bazel workspace right?

devversion commented on devtools/.bazelversion:
```diff
@@ -0,0 +1 @@
+4.0.0
```
Same for the two files above (can be removed?)

devversion commented on devtools/.npmrc:
```diff
@@ -0,0 +1 @@
+scripts-prepend-node-path=true
```
Nit: I think this can be removed as well?

devversion commented on devtools/projects/demo-no-zone/.bazelignore:
```diff
@@ -0,0 +1,3 @@
+node_modules
+dist
+bazel-out
```
I think this file can be removed?

devversion commented on devtools/projects/demo-no-zone/src/BUILD.bazel:
```diff
@@ -0,0 +1,62 @@
+    deps = [
+        ":devapp",
+    ],
+)
```
+ 💯 for using the custom devserver. I wanted to move away from the concatjs one regardless!

AleksanderBodurri replied:
Created a README in the devserver directory to note that it was duped from angular/components

devversion commented on devtools/projects/ng-devtools/src/lib/devtools-tabs/profiler/timeline/recording-visualizer/BUILD.bazel:
```diff
@@ -0,0 +1,56 @@
+        "timeline-visualizer.component.scss",
+        "tree-map-visualizer.component.scss",
+    ],
+)
```
For a follow-up to track: We should remove this target in favor of individual `sass_binary` targets. The multi sass binary rule is not very efficient and not used elsewhere in Angular. More details here: https://github.com/bazelbuild/rules_sass#multi_sass_binary.

We can have a custom macro that will build up multiple `sass_binary` directly, although in general explicit targets are preferred for explicit deps etc.

devversion commented on devtools/projects/ng-devtools/src/lib/vendor/angular-split/lib/BUILD.bazel:
```diff
@@ -0,0 +1,28 @@
+
+sass_binary(
+    name = "split_component_styles",
+    src = "//devtools/projects/ng-devtools/src/lib/vendor/angular-split/lib/component:split.component.scss",
```
If `lib/component` is its own Bazel package, why is the sass binary declared outside? (symptomatic for the inconveniently large label here). We could clean this up unless you had a reason for that.

devversion commented on devtools/tools/esbuild/esbuild-spec.config.mjs:
```diff
@@ -0,0 +1,37 @@
+  const {createLinkerEsbuildPlugin} = await import(
+    '@angular/dev-infra-private/shared-scripts/angular-linker/esbuild-plugin.mjs'
+  );
+  return await createLinkerEsbuildPlugin(/.*/, /* ensureNoPartialDeclaration */ true, { unknownDeclarationVersionHandling: 'ignore' });
```
Can you please add a comment on why we ignore version errors? (capturing the messages Joost basically put together I think)

devversion commented on devtools/tools/esbuild/BUILD.bazel:
```diff
@@ -0,0 +1,28 @@
+package(default_visibility = ["//visibility:public"])
+
+esbuild_config(
+    name = "esbuild_config",
```
Can this name be more specific? e.g. `esbuild_config_esm` or something? It's unclear what this config targets/what is is used for.

devversion commented on devtools/tools/esbuild/esbuild-iife.config.mjs:
```diff
@@ -0,0 +1,14 @@
+  resolveExtensions: ['.mjs', '.js'],
+  format: 'iife',
+  mainFields: ['es2020', 'es2015', 'module', 'main'],
+  conditions: ['es2020', 'es2015', 'module'],
```
You may also want to have the base options in a base config `.mjs` file, instead of repeating the resolve fields/extensions. three times?

devversion commented on package.json:
```diff
@@ -34,7 +35,13 @@
+    "devtools:test": "bazelisk test --config snapshot //devtools/...",
+    "devtools:e2e": "start-server-and-test devtools:devserver http-get://localhost:4200 cy:run",
+    "cy:run": "$(yarn bin)/cypress run --config-file=devtools/cypress.json",
+    "cy:open": "$(yarn bin)/cypress open --config-file=devtools/cypress.json"
```
Can these two scripts be more specific to the devtools (in terms of naming?). Also I think you can just use `cypress` directly as Yarn will expose `.bin` in the `PATH`

devversion commented on tslint.json:
```diff
@@ -139,7 +141,12 @@
+
+      "./devtools/dist/**/*",
+      "./devtools/bazel-bin/**/*",
+      "./devtools/bazel-out/**/*",
```
will these folders ever exist? I think bazel-bin/out will not since we have a single Bazel workspace in this repo

devversion commented on tools/BUILD.bazel:
```diff
@@ -38,3 +38,12 @@ platform(
 )
+
+nodejs_binary(
+    name = "tsc_wrapped_with_angular",
```
Unless I miss anything, I think this is no longer used right?

devversion commented on packages/compiler-cli/BUILD.bazel:
```diff
@@ -133,6 +133,7 @@ pkg_npm(
     # Do not add more to this list.
     # Dependencies on the full npm_package cause long re-builds.
     visibility = [
+        "//devtools/tools/esbuild:__pkg__",
```
I see many cases where you change the visibility of the `npm_package` targets. I think this was a leftover from a previous attempt right? If so, can you please try cleaning this up in all of `packages/..`?

devversion reviewed (approved):
LGTM overall. Left a few comments, but overall this looks good to me.

Note that we will need to re-audit how tests are executed, ensuring everything runs as desired, but I don't have sufficient context what tests we want to run as part of the devtools folder. Currently I just see that `devtools/...` runs, but there might be some tests outside Bazel not running I assume.

AleksanderBodurri committed:
refactor(devtools): disable cypress and remove dependencies

Previously we were running Cypress in CI. Since cypress needs to be installed to run these tests, it was decided that we would not run devtools' cypress tests until a decision could be made on an e2e testing solution.

This commit removes all package scripts, dependencies, and CI commands that reference cypress. Notably it leaves the actual tests in the devtools directory intact, so that they can be referenced later once a decision is made.

josephperrott commented on packages/bazel/src/ng_module/ng_module.bzl:
```diff
@@ -208,6 +208,7 @@ def _ngc_tsconfig(ctx, files, srcs, **kwargs):
     generate_ve_shims = _generate_ve_shims(ctx)
     compilation_mode = _get_ivy_compilation_mode(ctx)
     is_devmode = "devmode_manifest" in kwargs
+
```
I think we can undo this change.

mgechev commented on devtools/LICENSE:
```diff
@@ -0,0 +1,21 @@
+The MIT License
```
Sounds good to me.

AleksanderBodurri committed:
ci: sort devtools pullapprove reviewers in alphabetical order

Previously this list was not in any order. This commit puts the reviewer list in alphabetical order to be consistent with how other groups order reviewers

AleksanderBodurri committed:
refactor(devtools): remove LICENSE file in the devtools directory

This file was carried over from the original angular devtools repo. Now that devtools has been merged into angular/angular, this file is no longer necessary, because the root of this repo has its own LICENSE file.

AleksanderBodurri committed:
docs(devtools): update comment to better reflect what angular version is running in g3.

josephperrott reviewed (approved):
LGTM

AndrewKushnir commented:
FYI adding the "cleanup" label since there is a merge conflict with the master branch.

AleksanderBodurri commented:
> FYI adding the "cleanup" label since there is a merge conflict with the master branch.

Resolved, CI is running now

mgechev commented:
@AleksanderBodurri resolved the merge conflicts. Just removed the label. Should be ready to go 🚀

josephperrott commented:
This PR was merged into the repository by commit 8fb3aed.
